Makefile

```diff
@@ -27,24 +27,45 @@ all: check \
 dist/pyodide_py.tar: $(wildcard src/py/pyodide/*.py)  $(wildcard src/py/_pyodide/*.py)
 	cd src/py && tar --exclude '*__pycache__*' -cf ../../dist/pyodide_py.tar pyodide _pyodide
 
-dist/pyodide.asm.js: \
+src/core/pyodide_pre.o: src/js/_pyodide.out.js src/core/pre.js
+	rm -f tmp.dat
+	echo '()<::>{' >> tmp.dat
+	cat src/js/_pyodide.out.js >> tmp.dat
+	echo '}' >> tmp.dat
+	cat src/core/pre.js >> tmp.dat
+	echo "pyodide_js_init();" >> tmp.dat
+
+	rm -f src/core/pyodide_pre.gen.c
+	echo '__attribute__((used)) __attribute__((section("em_js"), aligned(1)))' >> src/core/pyodide_pre.gen.c
+	echo 'char __em_js__pyodide_js_init[] = {'  >> src/core/pyodide_pre.gen.c
+	cat tmp.dat  | xxd -i - >> src/core/pyodide_pre.gen.c
+	echo ', 0};' >> src/core/pyodide_pre.gen.c
+
+	rm tmp.dat
+	emcc -c src/core/pyodide_pre.gen.c -o src/core/pyodide_pre.o
+
+dist/libpyodide.a: \
```

Comment:
Do we want to include this in the dist archive?

Reply:
Not clear. I'm not sure where else to put it. We want it:
* as a separate github release
* in the circleci artifacts

Let's put it in dist for now and we can move it later when we figure out where to put it.

Reply:
> Let's put it in dist for now and we can move it later when we figure out where to put it.

Sounds good to me.